Specs/Core/sampleTerrainSpec.js

```diff
@@ -171,43 +171,49 @@ describe("Core/sampleTerrain", function () {
         deferred,
         overrideMimeType
       ) {
-        var sourceUrl = new URL(url, "https://google.com");
-        var sourceUrlPath =
-          sourceUrl.pathname + sourceUrl.search + sourceUrl.hash;
+        // find a key (source path) path in the spec which matches (ends with) the requested url
+        var availablePaths = Object.keys(proxySpec);
+        var proxiedUrl = null;
```

Comment:
Do `var proxiedUrl;` and down below `if (!defined(proxiedUrl)) {` (will need to include `Source/Core/defined.js`). That's the more standard way of handling null/undefined in the repo.